numba/tests/test_profile_refcount.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import division, absolute_import, print_function
+
+import numba.unittest_support as unittest
+import numpy as np
+from numba import jit
+from .support import MemoryLeakMixin, TestCase, tag
+import cProfile as profiler
+import sys
+
+
+class TestProfileRefCount(MemoryLeakMixin, unittest.TestCase):
```

Comment:
After I modified the patch to pass 'locals' as a function argument rather than keep it as a Dispatcher member this test didn't fail any longer, so it isn't good enough to measure correctness, unfortunately. I'm still chasing for a) the real issue and b) a good indicator for the issue's presence.